grails-app/controllers/org/pih/warehouse/user/UserController.groovy

```diff
@@ -266,32 +267,31 @@ class UserController {
 
 
     def disableLocalizationMode = {
+        // If we are in localization mode and want to change the language through the footer on React side
+        // we want to make sure we set the language to the clicked one, not for the previous locale
+        // we want to set the prev locale only if we disable localization mode through "Disable localization mode" button
+        Locale localeToSet = params?.localeToSet ? new Locale(params.localeToSet) : session?.prevLocale
```

Comment:
Let's just call this request parameter and variable "locale" (not localeToSet) since that's what it's been called. I'm worried that adding a new request parameter will cause confusion.

Reply:
For now it is only used for React part, as I didn't have an idea how to determine whether I disable localization mode through button "Disable localization mode" or I  just change the language.
When disabling the localization mode through button "Disable localization mode" we want to go back to previous chosen locale.
When changing the language while being in localization mode, we want to disable localization mode, but we don't want to go back to previous chosen locale, but to change it for the language we've just clicked, so that's why I had to figure out some way to determine this.
For GSP part it's unnecessary, because we always change the language through `updateUserAuthLocale`.
For React part I just add something like this when changing the language being in localization mode:
```
 href={`/openboxes/user/disableLocalizationMode?localeToSet=${language.code}`}
```
I know it may cause confusion, but I couldn't figure out anything better and that's why I make the checking if this param is provided and that's why I added a comment above this.
If you have any idea how I could deal with this case, I'd love to hear from you